### PR TITLE
cmakeWithQt4Gui: drop

### PR DESCRIPTION
**Pull request by alyssais**
###### Motivation for this change

Support for Qt 4 was [removed](https://cmake.org/cmake/help/latest/release/3.19.html#deprecated-and-removed-features) in CMake 3.19.0.

ZHF: #122042

@NixOS/nixos-release-managers

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/development/tools/build-managers/cmake/default.nix [hunk 1]
```diff
@@ -7,19 +7,14 @@
...
 
-assert withQt5 -> useQt4 == false;
-assert useQt4 -> withQt5 == false;
-
 stdenv.mkDerivation (rec {
   pname = "cmake"
           + lib.optionalString isBootstrap "-boot"
           + lib.optionalString useNcurses "-cursesUI"
           + lib.optionalString withQt5 "-qt5UI"
-          + lib.optionalString useQt4 "-qt4UI";
```
Comment (jtojnar) [inline pkgs/development/tools/build-managers/cmake/default.nix:22]: Missing semicolon.

File: pkgs/top-level/all-packages.nix [hunk 2]
```diff
@@ -12762,7 +12762,6 @@ in
   cmakeCurses = cmake.override { useNcurses = true; };
 
   cmakeWithGui = cmakeCurses.override { withQt5 = true; };
-  cmakeWithQt4Gui = cmakeCurses.override { useQt4 = true; };
```
Comment (SuperSandro2000) [inline pkgs/top-level/all-packages.nix:12765]: Missing an entry to aliases.nix.

**Comment by ajs124** [comment]
This should also tick a box or two in #33248

**Comment by jonringer** [comment]
an issue from 4 years ago... >.>

File: pkgs/top-level/all-packages.nix (hunk 2 above)
Comment (jonringer) [reply pkgs/top-level/all-packages.nix:12765]: bump. Please add a `cmakeWithQt4Gui = throw "..."` entry in aliases.nix

**Pull request merged**